devel/retdec/Portfile

```diff
@@ -43,6 +43,13 @@ if {${name} eq ${subport}} {
     # See https://github.com/avast/retdec/pull/807
     openssl.branch  1.1
 
+    if { ${build_arch} eq {arm64} } {
+        known_fail yes
+        pre-fetch {
+            ui_error "${name} can't be build for arm64, use ${name}-devel port instead."
+            return -code error "Unsupported architecture"
+        }
+    }
```

Comment:
isn't this what `supported_archs` is for?

Reply:
@reneeotten `supported_archs i386 ppc ppc64 x86_64` triggers to build dependencies whcih includes `graphviz` with `+universal` which is failed on https://trac.macports.org/ticket/63561 but it replaces tons of ports to `+universal` variant.

Reply:
I don't see how that has anything to do with this. If you cannot build for `arm64` but it should build for all other architectures, I the line you mention seems correct. If that doesn't work there must be something else going on...

Reply:
@reneeotten let me try to fix `dav1d`...

Reply:
I don't see why that would be necessary if you're effectively just removing `arm64` from the lists of supported_archs. If it doesn't work then, it shouldn't work now either on other architectures, unless I am misunderstanding something here... admittedly I don't fully understand how this universal builds stuff is done... What happens if you add `universal_variant  no`?

Reply:
@reneeotten it still tries to rebuild everything for `+universal`. I'm fixing a few trivial port, but I afraid that it may trigger something huge.

Reply:
then I'd suggest to just comment in that Trac ticket and/or wait for someone who understands this universal stuff to stop by at this PR and educate us (or at least me) on what's going on

Reply:
@reneeotten this is expected to be honest. M1 can run both x86_64 and arm64 application, and if application can't be build for arm64 but can be built for x86_64 => it tries to build this.

Is need to rebuild all dependencies with `+universal` which isn't `noarch` and if it can => just build x86_64 version.

Reply:
Ok, I gave up. It tries to build `rust` / `cargo` as `+universal` to build some library... and `rust` port defently not something which can be made `+universal` by some simple hack.